projects/observability/src/shared/dashboard/data/graphql/specifiers/attribute-sort-specification.model.ts

```diff
@@ -0,0 +1,33 @@
+import { Dictionary } from '@hypertrace/common';
+import { Model, ModelProperty, STRING_PROPERTY } from '@hypertrace/hyperdash';
+import { GraphQlSortDirection } from '../../../../graphql/model/schema/sort/graphql-sort-direction';
+import { Specification } from '../../../../graphql/model/schema/specifier/specification';
+import { SpecificationBuilder } from '../../../../graphql/request/builders/specification/specification-builder';
+import { SpecificationModel } from './specification.model';
+
+@Model({
+  type: 'attribute-sort-specification',
+  displayName: 'Attribute Sort'
+})
+export class AttributeSortSpecificationModel extends SpecificationModel<Specification> {
+  @ModelProperty({
+    key: 'key',
+    type: STRING_PROPERTY.type,
+    required: true
+  })
+  public key!: string;
+
+  @ModelProperty({
+    key: 'direction',
+    type: STRING_PROPERTY.type,
```

Comment:
we need to mark it as required if it is

Reply:
direction could be undefined IMO (when is not applied). Do you think that should be mandatory anyways?